### PR TITLE
`local-exec` documentation lacks on-destroy documentation

### PR DESCRIPTION
**Pull request by miyagawa-michiru**
# Issue
https://github.com/hashicorp/terraform/issues/30953#issue-1218566474

> The documentation regarding local-exec lacks documentation regarding resource destruction.

I added a little bit about the destruction option for  local-exec because there was no mention about it.

### CODE REVIEW

**Review by joao-luna-98 — APPROVED** [review]
Looks good to me.

File: website/docs/language/resources/provisioners/local-exec.mdx [hunk 1]
```diff
@@ -57,6 +57,12 @@ The following arguments are supported:
 * `environment` - (Optional) block of key value pairs representing the
   environment of the executed command. inherits the current process environment.
 
+* `when` - (Optional) If provided, specifies the timing when the command will
+  be executed. If `when = destroy` is specified, the provisioner will run when
+  the resource it is defined within is destroyed. See
+  [Destroy-Time Provisioners](/language/resources/provisioners/syntax#destroy-time-provisioners)
+  for more details.
```
Comment (laurapacilio) [inline website/docs/language/resources/provisioners/local-exec.mdx:64]: Just a nit to make the language more inclusive and remove a few extra words. Thank you!

```suggestion
* `when` - (Optional) If provided, specifies Terraform will execute the command. For example, `when = destroy` specifies that the provisioner will run when
  the associated resource is destroyed. Refer to
  [Destroy-Time Provisioners](/language/resources/provisioners/syntax#destroy-time-provisioners)
  for details.
```

**Comment by joao-luna-98** [comment]
Yes I am, it goes with what I was expecting to see. :)

File: website/docs/language/resources/provisioners/local-exec.mdx (hunk 1 above)
Comment (miyagawa-michiru) [reply website/docs/language/resources/provisioners/local-exec.mdx]: I got it. Thank you for your review. I'll fix it.

**Comment by miyagawa-michiru** [comment]
@laurapacilio 
I would appreciate if you could review it. 
https://github.com/hashicorp/terraform/pull/31049/commits/2bb83244f9eabf7907035592f289487e5cf1b28e

File: website/docs/language/resources/provisioners/local-exec.mdx [hunk 2]
```diff
@@ -57,6 +57,11 @@ The following arguments are supported:
 * `environment` - (Optional) block of key value pairs representing the
   environment of the executed command. inherits the current process environment.
 
+* `when` - (Optional) If provided, specifies Terraform will execute the command.
```
Comment (laurapacilio) [inline website/docs/language/resources/provisioners/local-exec.mdx:60]:
```suggestion
* `when` - (Optional) If provided, specifies when Terraform will execute the command.
```

**Review by laurapacilio — CHANGES_REQUESTED** [review]
Sorry! I realized there is one word missing. After you merge this in, I will merge your PR and close out the associated issue.

Thank you again for your work on this!

File: website/docs/language/resources/provisioners/local-exec.mdx (hunk 2 above)
Comment (miyagawa-michiru) [reply website/docs/language/resources/provisioners/local-exec.mdx]: I fixed it. Thank you for your review.

**Review by joao-luna-98 — APPROVED** [review]

**Review by laurapacilio — APPROVED** [review]

**Pull request merged**